engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnStateTransitionBehavior.java

```diff
@@ -75,35 +74,33 @@ public BpmnStateTransitionBehavior(
 
   /** @return context with updated intent */
   public BpmnElementContext transitionToActivating(final BpmnElementContext context) {
-    if (MigratedStreamProcessors.isMigrated(context.getBpmnElementType())) {
-      final var elementInstance = stateBehavior.getElementInstance(context);
-      if (elementInstance != null) {
-        verifyIncidentResolving(context, "#transitionToActivating");
-        // if the element already exists, then the Activate_Element command is processed as a result
-        // of resolving an incident. We don't have to transition again. Just update the context
-        return context.copy(
-            context.getElementInstanceKey(),
-            context.getRecordValue(),
-            ProcessInstanceIntent.ELEMENT_ACTIVATING);
-      }
 
-      // When the element instance key is not set (-1), then we process the ACTIVATE_ELEMENT
-      // command. We need to generate a new key in order to transition to ELEMENT_ACTIVATING, such
-      // that we can assign the new create element instance a correct key. It is expected that on
-      // the command the key is not set. But some elements (such as multi instance), need to
-      // generate the key before they write ACTIVATE command, to prepare the state (e.g. set
-      // variables) for the upcoming element instance.
-      if (context.getElementInstanceKey() == -1) {
-        final var newElementInstanceKey = keyGenerator.nextKey();
-        final var newContext =
-            context.copy(
-                newElementInstanceKey,
-                context.getRecordValue(),
-                ProcessInstanceIntent.ELEMENT_ACTIVATING);
-        return transitionTo(newContext, ProcessInstanceIntent.ELEMENT_ACTIVATING);
-      }
+    final var elementInstance = stateBehavior.getElementInstance(context);
+    if (elementInstance != null) {
+      verifyIncidentResolving(context, "#transitionToActivating");
+      // if the element already exists, then the Activate_Element command is processed as a result
+      // of resolving an incident. We don't have to transition again. Just update the context
+      return context.copy(
+          context.getElementInstanceKey(),
+          context.getRecordValue(),
+          ProcessInstanceIntent.ELEMENT_ACTIVATING);
+    }
+
+    var transitionContext = context;
```

Comment:
🙃 Please consider making this [local variable final](https://github.com/camunda-cloud/zeebe/wiki/Code-Style#favor-final-modifier). You could for example, add an else branch to the if statement below.